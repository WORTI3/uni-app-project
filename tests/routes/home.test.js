const request = require('supertest');
const cheerio = require('cheerio');

const app = require('../../src/app');

const url = '/';

describe('GET ' + url, () => {
  it('should show the portal home page when no user present', async () => {
    const result = await request(app)
      .get(url)
      .expect(200);

    expect(result.get('Content-Type')).toContain('text/html');
    expect(result.text).toMatchSnapshot();
    expect(result.body.user).toBeUndefined();
  });

  it('should render the home page when user is not authenticated', async () => {
    const result = await request(app)
      .get(url)
      .expect(200);

    const $ = cheerio.load(result.text);
    const heading = $('h1').text();
    expect(heading).toBe('Welcome to the portal');
    const subHeading = $('h3').text();
    expect(subHeading).toBe('to access the portal you must have an account');
    expect($('a').length).toBe(4);
  });
});

