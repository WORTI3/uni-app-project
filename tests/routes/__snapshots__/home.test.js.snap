// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET / should show the portal home page when no user present 1`] = `
"<!DOCTYPE html>
<html lang="en">
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1">
		<title>Home • wortie</title>
		<link rel="stylesheet" type="text/css" href="/main.css">
	</head>

    <div class="flex flex-col h-screen justify-between overflow-hidden">
        <nav class="flex items-center justify-between flex-wrap bg-gray-100 p-6">
    <div class="flex items-center flex-shrink-0 text-picton-blue-600 mr-6">
        <span class="font-bold text-xl tracking-tight font-sans">WORTIE</span>
    </div>
    <div class="block lg:hidden">
        
    </div>
    <div class="flex items-center w-auto">
        
        <div class="text-sm">
            </div>
            
            <div>
                <a href="/login" class="px-4 py-2 border bg-picton-blue-500 text-white font-bold text-xs leading-tight uppercase rounded shadow-md hover:bg-spicton-blue-700 hover:shadow-lg focus:bg-picton-blue-900 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-picton-blue-900 active:shadow-lg transition duration-150 ease-in-out">Sign in</a>
            </div>
        
    </div>
</nav>

        <body class="bg-gradient-to-tr from-picton-blue-400 to-picton-blue-700 overflow-hidden">
            <main>
<div class="text-center text-white py-20 px-6">
    <h1 class="text-5xl font-bold mt-0 mb-6">Welcome to the portal</h1>
    <h3 class="text-xl font-semibold mb-8">to access the portal you must have an account</h3>
	<a class="inline-block px-6 py-2.5 bg-picton-blue-800 text-white font-bold text-xs leading-tight uppercase rounded shadow-md hover:bg-picton-blue-700 hover:shadow-lg focus:bg-picton-blue-900 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-picton-blue-900 active:shadow-lg transition duration-150 ease-in-out" data-mdb-ripple="true" data-mdb-ripple-color="light" href="/login" role="button">Sign in</a>
    <p class="inline px-5"> or </p>
    <a class="inline-block px-6 py-2.5 bg-picton-blue-800 text-white font-bold text-xs leading-tight uppercase rounded shadow-md hover:bg-picton-blue-700 hover:shadow-lg focus:bg-picton-blue-900 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-picton-blue-900 active:shadow-lg transition duration-150 ease-in-out" data-mdb-ripple="true" data-mdb-ripple-color="light" href="/signup" role="button">Sign up</a>
</div>
</main>
        </body>

        <footer class="lg:text-left bg-gray-100 text-gray-600">
            <div class="text-center p-6">
                <span>© 2023 </span>
                <a class="text-gray-600 font-semibold" href="https://github.com/WORTI3">Wortie</a>
            </div>
        </footer>
    </div>
</html>"
`;
